metrics-core/src/main/java/com/codahale/metrics/CsvReporter.java

```diff
@@ -180,8 +202,13 @@ private CsvReporter(MetricRegistry registry,
         super(registry, "csv-reporter", filter, rateUnit, durationUnit, executor, shutdownExecutorOnStop);
         this.directory = directory;
         this.locale = locale;
+        this.separator = separator;
         this.clock = clock;
         this.csvFileProvider = csvFileProvider;
+
+        this.histogramFormat = String.join(separator, "%d","%d","%f","%d","%f","%f","%f","%f","%f","%f","%f");
```

Comment:
While it was nice to have the headers and the format right next to each other in the code, there isn't a need to construct the same value over and for each call to `report`